src/NHibernate/Async/Type/AnyType.cs

```diff
@@ -0,0 +1,134 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Collections;
+using System.Data.Common;
+using System.Reflection;
+using NHibernate.Engine;
+using NHibernate.Persister.Entity;
+using NHibernate.Proxy;
+using NHibernate.SqlTypes;
+using NHibernate.Util;
+using System.Collections.Generic;
+
+namespace NHibernate.Type
+{
+	using System.Threading.Tasks;
+	using System.Threading;
+	/// <content>
+	/// Contains generated async methods
+	/// </content>
+	public partial class AnyType : AbstractType, IAbstractComponentType, IAssociationType
+	{
+
+		public override Task<object> NullSafeGetAsync(DbDataReader rs, string name, ISessionImplementor session, object owner, CancellationToken cancellationToken)
+		{
+			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<object>(cancellationToken);
+			}
+			throw new NotSupportedException("object is a multicolumn type");
```

Comment:
This is a bug in the generator, related [issue](https://github.com/maca88/AsyncGenerator/issues/4). 
Currently, the generator is not checking if the method is faulted when adding a cancellation token guard.